src/interfaces/vault.ts

```diff
@@ -443,7 +443,7 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
       combinedParams.gasPrice = undefined;
       return await this.yearn.services.transaction.sendTransaction(combinedParams);
     } catch (error) {
-      if ((error as any).code === -32602) {
+      if ((error as { code: number }).code === -32602) {
```

Comment:
It would be nice to have an actual `Error` type that has a `code` prop

Reply:
Yeah I think we get this error from ethers, not sure if there's a type for it

Reply:
Had a look at the library and I think they are internally aware of the issue as they have the `// @TODO: Any??` comment in there :) -- so I'm guessing there's no Error type coming from there

```typescript
    makeError(message: string, code?: ErrorCode, params?: any): Error {
        // ...

        // @TODO: Any??
        const error: any = new Error(message);
        error.reason = reason;
        error.code = code

        Object.keys(params).forEach(function(key) {
            error[key] = params[key];
        });

        return error;
    }
```